### PR TITLE
Not to tag versioned transaction as simple_vote_tx

### PR DESCRIPTION
**Pull request by taozhu-chicago**
#### Problem
Versioned vote transaction should not be tagged as _simple_ vote tx. 

#### Summary of Changes
- filter versioned transaction out of `check_for_simple_vote_transaction`

Fixes #

### CODE REVIEW

File: perf/src/sigverify.rs [hunk 1]
```diff
@@ -358,6 +358,17 @@ fn check_for_simple_vote_transaction(
         return Err(PacketError::InvalidSignatureLen);
     }
 
+    // simple vote should only be legacy message
+    let msg_start = (packet_offsets.msg_start as usize)
+        .checked_sub(current_offset)
+        .ok_or(PacketError::InvalidLen)?;
+    let message_prefix = *packet
+            .data(msg_start)
+            .ok_or(PacketError::InvalidSignatureLen)?;
```
Comment (t-nelson) [inline perf/src/sigverify.rs:367]: `PacketError::InvalidLen`?  signature seems like a misnomer

**Review by t-nelson — APPROVED** [review]
thanks!  r+ suggested error type switch

**Review by t-nelson — DISMISSED** [review]
:shipit:

**Comment by mvines** [comment]
Does `is_simple_vote_transaction` need an update as well?
https://github.com/solana-labs/solana/blob/45a34f637b7905e642bdc979c2a646aed24a184f/runtime/src/vote_parser.rs#L16

**Comment by mvines** [comment]
This also seems broken? https://github.com/solana-labs/solana/blob/45a34f637b7905e642bdc979c2a646aed24a184f/sdk/src/transaction/sanitized.rs#L119-L123
It checks that the first instruction is for the vote program but nothing else

**Comment by t-nelson** [comment]
both are downstream of sigverify, right?  should probably just use the flag set there for all of these

**Comment by mvines** [comment]
> both are downstream of sigverify, right? should probably just use the flag set there for all of these

Should be, so yes the other sites could be cleaned up in a follow-up for master.

**Comment by taozhu-chicago** [comment]
> Should be, so yes the other sites could be cleaned up in a follow-up for master.

Thank you for catching these inconsistencies. Just created #30521 to consolidate is_simple_vote calls.

**Comment by taozhu-chicago** [comment]
@t-nelson can you re-approve it if OK to address the other call sites separately?

**Review by t-nelson — APPROVED** [review]
:saluting-face:

**Pull request merged**